compiler/rustc_smir/src/rustc_smir/context.rs

```diff
@@ -587,11 +587,20 @@ impl<'tcx> Context for TablesWrapper<'tcx> {
         }
     }
 
+    /// Retrieve the plain intrinsic name of an instance if it's an intrinsic.
+    ///
+    /// The plain name does not include type arguments (as `instance_name`
+    /// does), which is more convenient to match with intrinsic symbols.
     fn intrinsic_name(&self, def: InstanceDef) -> Option<Symbol> {
         let tables = self.0.borrow_mut();
         let instance = tables.instances[def];
-        let name = tables.tcx.intrinsic(instance.def_id())?;
-        Some(name.to_string())
+        if !(matches!(instance.def, ty::InstanceDef::Intrinsic(..))
+            || matches!(instance.def, ty::InstanceDef::Item(..)))
```

Comment:
Flip the polarity of the `if` statement and put the `Some` arm into the `if`, rather than using an early return, and merge the `matches!()` into one. Or better yet, use an exhaustive match to make sure people don't accidentally forget to add their instances to this list.

```suggestion
        if matches!(instance.def, ty::InstanceDef::Intrinsic(..) | ty::InstanceDef::Item(..))
```

Reply:
Replaced this with an exhaustive match.